Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,199 @@
+            { 
+                return CreateJobs(request, response); 
+            };
+
+        sdfAssetBuilderDescriptor.m_processJobFunction = [this](auto && request, auto && response) 
+            { 
+                return ProcessJob(request, response); 
+            };
+
+        // Listen for asset builder notifications requesting jobs for any of the sdf source file types.
+        BusConnect(sdfAssetBuilderDescriptor.m_busId);
+
+        // Register this builder with the AssetBuilderSDK.
+        AssetBuilderSDK::AssetBuilderBus::Broadcast(
+            &AssetBuilderSDK::AssetBuilderBus::Handler::RegisterBuilderInformation, sdfAssetBuilderDescriptor);
+    }
+
+    SdfAssetBuilder::~SdfAssetBuilder()
+    {
+        // Stop listening for asset builder notifications.
+        BusDisconnect();
+
+        // The AssetBuilderSDK doesn't support deregistration, so there's nothing more to do here.
+    }
+
+    void SdfAssetBuilder::CreateJobs(
+        const AssetBuilderSDK::CreateJobsRequest& request,
+        AssetBuilderSDK::CreateJobsResponse& response) const
+    {
+        // For each input file, create an output job.
+        for (const AssetBuilderSDK::PlatformInfo& platformInfo : request.m_enabledPlatforms)
+        {
+            AssetBuilderSDK::JobDescriptor jobDescriptor;
+            jobDescriptor.m_critical = false;
+            jobDescriptor.m_jobKey = "SDF (Simulation Description Format) Asset";
+            jobDescriptor.SetPlatformIdentifier(platformInfo.m_identifier.c_str());
+
+            response.m_createJobOutputs.push_back(jobDescriptor);
+        }
+
+        response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+    }
+
+    void SdfAssetBuilder::ProcessJob(
+        const AssetBuilderSDK::ProcessJobRequest& request,
+        AssetBuilderSDK::ProcessJobResponse& response) const
+    {
+        // The following code is just a stub while building the SdfAssetBuilder. 
+        // Right now, it just generates an empty default procedural prefab. It still needs to be modified
+        // to create a procedural prefab that contains the correct information in it.
+
+        AZStd::string outputFilename = request.m_sourceFile;
+        AzFramework::StringFunc::Path::ReplaceExtension(outputFilename, "procprefab");
+
+        AZStd::string tempAssetOutputPath;
+        AzFramework::StringFunc::Path::ConstructFull(request.m_tempDirPath.c_str(), outputFilename.c_str(), tempAssetOutputPath, true);
+
+        AZStd::string prefabJson = CreateDefaultProcPrefab(request, response);
+
+        // If no prefab string was generated, then return a failure.
+        if (prefabJson.empty())
+        {
+            response.m_resultCode = AssetBuilderSDK::ProcessJobResult_Failed;
+            return;
+        }
+
+        // Save the prefab to our output directory.
+        rapidjson::Document doc;
+        doc.Parse(prefabJson.c_str());
+        AZ::JsonSerializationUtils::WriteJsonFile(doc, tempAssetOutputPath.c_str());
```

Comment:
Shouldn't the `WriteJsonFile` outcome be checked for errors and the error message logged from it?

Reply:
Technically yes, but I'm expecting this part of the code to change pretty drastically over the next couple of PRs. I thought it was pretty likely that there would be different / better ways to write out prefabs than this, so I expected this to just be a stub. I'll add the error message logging anyways though, it's always a good idea to keep each PR as correct as possible. :)